### PR TITLE
Improve Scio on Jupyter

### PR DESCRIPTION
**Pull request by anish749**
I tried to make this more usable.

Since this was almost 2 yrs old lying un used, I took the liberty to break some of the APIs.

TL;DR:
* upgrade to scio 0.6.1
* Add api to easily create multiple Scio context with same PipelineOptions.
* Some helper functions to make use of Scio easier from Jupyter.

This primarily adds the functionality to easily close and recreate Scio contexts with same pipeline options and run on data flow / other runners. It still doesn't make things very interactive as outputs are still not in-memory, and its takes a few minutes to start a Dataflow Job.
This PR however makes it easy to iteratively develop batch pipelines in Scio / Beam.

Also we can use Taps to temporarily materialize a SCollection to the staging bucket and read the data from there. This makes analysis somewhat easier.

### CODE REVIEW

**Comment by anish749** [comment]
Hey @alexarchambault would you kindly review this PR..

**Comment by alexarchambault** [comment]
Hey @anish749, sorry for the delay.

I can merge, then make a release, if you have an immediate use of this. But most development now happens on the [`develop` branch](https://github.com/jupyter-scala/jupyter-scala/tree/develop). It supports spark via [this project](https://github.com/alexarchambault/ammonite-spark) that targets Ammonite, that it just [extends a bit](https://github.com/jupyter-scala/jupyter-scala/tree/develop/modules/scala/almond-spark) to get some extra Jupyter-specific niceties (progress bars, …).

For scio, I guess a project similar to [ammonite-spark](https://github.com/alexarchambault/ammonite-spark) could be written, adding scio support to Ammonite. It can then be used as is from the [upcoming version](https://github.com/jupyter-scala/jupyter-scala/tree/develop) of jupyter-scala. I was thinking of maybe renaming ammonite-spark to something like ammonite-bigdata, and add support for scio in it, among other stuff. Is it something that would be useful for you?

**Comment by anish749** [comment]
Hey @alexarchambault I think the dev branch is quite ahead of the master. What is the plan for merging and release of the next version? If it is longer, then it would be great to have v0.4.3.

If the idea is to separate out the `jupyter-scala` repo into multiple repos, I feel it might be a good idea to have Scio as `ammonite-scio` separate from `ammonite-spark`, given that a majority of users who would plan to use Spark for interactive analysis would not be using Scio at the same time and vice versa. 

The problem with Beam / Scio is that it is not very well suited for interactive analysis at the moment, which narrows the use cases while in Jupyter. There were times when I really felt the need of having a notebook based environment, and hence started experimenting with this.

**Comment by alexarchambault** [comment]
@anish749 Sure, go for it for https://github.com/jupyter/docker-stacks, don't hesitate to ping me there for feedback.

FYI, @Atry already wrote and pushed a docker image for almond, see https://github.com/almond-sh/almond/issues/214#issue-354752234 (but it's not added to docker-stacks)

**Comment by Atry** [comment]
@anish749 You can find those docker images at https://hub.docker.com/r/popatry/almond-images/
